nipy/modalities/fmri/design_matrix.py

```diff
@@ -61,30 +61,42 @@ def _poly_drift(order, frametimes):
     return pol
 
 
-def _cosine_drift(hfcut, frametimes):
+def _cosine_drift(hfcut, tim):
     """Create a cosine drift matrix
 
     Parameters
     ----------
-    hfcut, float , cut frequency of the low-pass filter
-    frametimes: array of shape(nscans): the sampling time
+    hfcut, float 
+         Cut period of the low-pass filter (in sec)
+    tim: array of shape(nscans)
+         The sampling times (in sec)
 
     Returns
     -------
     cdrift:  array of shape(n_scans, n_drifts)
-             polynomial drifts plus a constant regressor
+             cosin drifts plus a constant regressor at cdrift[:,0]
+
+    Ref: http://en.wikipedia.org/wiki/Discrete_cosine_transform DCT-II
     """
-    tmax = float(frametimes.max())
-    tsteps = len(frametimes)
-    order = int(np.floor(2 * float(tmax) / float(hfcut)) + 1)
-    cdrift = np.zeros((tsteps, order))
-    for k in range(1, order):
-        cdrift[:, k - 1] = np.sqrt(2.0 / tmax) * np.cos(
-            np.pi * (frametimes / tmax + 0.5 / tsteps) * k)
-    cdrift[:, order - 1] = np.ones_like(frametimes)
+    N = len(tim)
```

Comment:
Capital letters should be used for constants. Rather use e.g. `len_sampling_times`